### PR TITLE
feat(sdk): Modify Interactive Question Layout

### LINKED ISSUE #49719: Improve usability of InteractiveQuestion's default view by using contextual querying components
After we have implemented the modular querying components such as `FilterSelector`, `FilterPanel` and `BreakoutSelector`, we want to use those components to improve the default question view (the `InteractiveQuestionResult`).

We can [follow the design](https://www.figma.com/design/W307FIujMcLXD7alT1M0qc/Improve-UX-for-self-service-analytics-using-SDK?t=Rt46sDH5WnBChwh3-0) here and use contextual querying components in place of full-screen components, such as `FilterSelector` in a popover for selecting filters. This makes the default querying experience better than switching between components that takes up the full screen.

See the [tech docs](https://www.notion.so/metabase/Tech-Improving-the-question-component-s-querying-experience-for-embedding-sdk-13169354c901802582a8ca9389c544bd?pvs=4#13769354c90180e98136c53dfbadd084) for more info.

### PR DESCRIPTION
**Pull request by oisincoveney**
Checklist for Oisin:
- check that [feat(sdk): revamp CreateQuestion and create question behaviour](https://github.com/metabase/metabase/pull/50088) still has the same functionality after

### CODE REVIEW

**Comment by oisincoveney** [comment]
This change is part of the following stack:

- #50152 ◀
    - #50509

<sub>Change managed by [git-spice](https://abhinav.github.io/git-spice/).</sub>

**Review by heypoom — COMMENTED** [review]
I just did some testing on Storybook and Shoppy with theming applied. It looks great so far! Just a couple minor theming things to fix later. I'll review the code itself tomorrow.

https://www.loom.com/share/ccd540de7c014e5fa0493743f5376bcb?sid=2e566612-2012-4c05-9670-28d668956900

File: enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/components/ChartTypeSelectorList/ChartTypeDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+          <Menu.Item
+            key={`${value}/${index}`}
+            onClick={() => updateQuestionVisualization(value)}
+            icon={iconName ? <Icon name={iconName} /> : null}
+          >
+            {label}
+          </Menu.Item>
+        ))}
+
+        <Menu.Label>{t`Nonsensible`}</Menu.Label>
```
[the remark this answers is not in the document]
Comment (oisincoveney) [reply enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/components/ChartTypeSelectorList/ChartTypeDropdown.tsx:82]: Good call, I really don't like that terminology. It doesn't strike me as either "Metabasey", simple, or easily indicative of what it really means.

[the remark this answers is not in the document]
Comment (mazameli) [reply enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/components/ChartTypeSelectorList/ChartTypeDropdown.tsx:82]: This should just say `Other charts`

**Review by heypoom — APPROVED** [review]
Functionality-wise this works great! ([see other comment on GitHub](https://github.com/metabase/metabase/pull/50152#pullrequestreview-2477894228))

PS. I noticed that the viz flashes for a brief moment (recorded in the Loom), but I'm not sure if that is on the viz side of things or some re-rendering issues in the sdk.

**Comment by WiNloSt** [comment]
2 things I notice. I think this is popover widget dependent and it'd require a lot of effort to go fix all of this, but let's clean them up as we find.

1. This kind of date filter still doesn't go back to the main filter list popover view. In the image 1 this happens after I click on an existing filter, then after I click back I got image 2.
    - Screenshot 2024-12-09 at 3 50 12 PM
    - Screenshot 2024-12-09 at 3 50 15 PM
2. didn't we use to have some kind of reset button? I just messed my question and there's no way to reset the question back to its original state, and I'm a little sad if I have to refresh the whole page for that.
Screenshot 2024-12-09 at 3 50 28 PM

**Comment by oisincoveney** [comment]
> 2 things I notice. I think this is popover widget dependent and it'd require a lot of effort to go fix all of this, but let's clean them up as we find.
> 
> 1. This kind of date filter still doesn't go back to the main filter list popover view. In the image 1 this happens after I click on an existing filter, then after I click back I got image 2.
>    
>    * Screenshot 2024-12-09 at 3 50 12 PM
>    * Screenshot 2024-12-09 at 3 50 15 PM

Yep, I should have mentioned in the last PR. This behavior is way harder to fix than it looks, since the date picker is really intertwined within a deep part of the code and changing the navigation of that will take more time than I think is useful right now. I can add it to the task list of things. 

What is the expected behavior again? If I click the back button, it does straight back to the badge list, right?


> 2. didn't we use to have some kind of reset button? I just messed my question and there's no way to reset the question back to its original state, and I'm a little sad if I have to refresh the whole page for that.
>    Screenshot 2024-12-09 at 3 50 28 PM

the reset should be there - it should pop up above the title (similar to the core app) as text that says "Back to [ORIGINAL QUESTION NAME]".  I'm using the same logic as the reset button so that's super weird.

File: enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/components/ChartTypeSelectorList/ChartTypeDropdown.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (oisincoveney) [reply enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/components/ChartTypeSelectorList/ChartTypeDropdown.tsx]: done

**Comment by oisincoveney** [comment]
@metabase-bot backport release-x.51.x

**Pull request merged**